### PR TITLE
Allow vmexport to have names longer than 63 characters

### PR DESCRIPTION
**Pull request by awels**
### What this PR does
Before this PR:
It was possible to create a VMExport resource with more than 63 characters, but the export would not succeed because the name of the export was used as part of the  name and label selector of the service. Since labels cannot be more than 63 character this would cause the creation of the service to fail.

The name of the container in the pod was also based on the VMExport resource, and that name cannot be more than 63 characters either.

It would also generate warnings in the log about the name of the export pod which is also based on the name of the VMExport resource was too long and that it could cause undesired behavior.

After this PR:
The service and pod are created properly if the vmexport name is longer than 63 characters. There are also no warnings in the log about the name of the export pod being too long and causing unwanted issues.

Fixes https://issues.redhat.com/browse/CNV-47008

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [X] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [X] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
BugFix: VMexport with names longer than 63 characters no longer breaks the export.
```

### CODE REVIEW

File: pkg/virt-controller/services/template.go [hunk 1]
```diff
@@ -1104,7 +1107,7 @@ func (t *templateService) RenderHotplugAttachmentTriggerPodTemplate(volume *v1.V
 func (t *templateService) RenderExporterManifest(vmExport *exportv1.VirtualMachineExport, namePrefix string) *k8sv1.Pod {
 	exporterPod := &k8sv1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      fmt.Sprintf("%s-%s", namePrefix, vmExport.Name),
+			Name:      naming.GetName(namePrefix, vmExport.Name, validation.DNS1035LabelMaxLength),
```
Comment (mhenriks) [inline pkg/virt-controller/services/template.go:1110]: `naming.GetPodName()` seems very appropriate here

File: pkg/virt-controller/services/template.go [hunk 2]
```diff
@@ -1121,7 +1124,7 @@ func (t *templateService) RenderExporterManifest(vmExport *exportv1.VirtualMachi
 			RestartPolicy: k8sv1.RestartPolicyNever,
 			Containers: []k8sv1.Container{
 				{
-					Name:            vmExport.Name,
+					Name:            naming.GetName("virt-export", vmExport.Name, validation.DNS1035LabelMaxLength),
```
Comment (mhenriks) [inline pkg/virt-controller/services/template.go:1127]: are container names limited?

**Comment by awels** [comment]
Closing in favor of https://github.com/kubevirt/kubevirt/pull/12659

File: pkg/virt-controller/services/template.go (hunk 2 above)
Comment (awels) [reply pkg/virt-controller/services/template.go:1127]: Yes, if the name is too long it won't pass the pod validation

**Pull request closed without merging**